pallets/process-validation/src/lib.rs

```diff
@@ -79,50 +100,113 @@ pub mod pallet {
 
     /// Storage map definition
     #[pallet::storage]
-    #[pallet::getter(fn processes_by_id_and_version)]
-    pub(super) type ProcessesByIdAndVersion<T: Config> = StorageDoubleMap<
+    #[pallet::getter(fn get_process)]
+    pub(super) type GetProcess<T: Config> = StorageDoubleMap<
         _,
         Blake2_128Concat,
         T::ProcessIdentifier,
         Blake2_128Concat,
-        T::ProcessVersion,
+        i32,
         Process,
+        ValueQuery,
+    >;
+
+    #[pallet::storage]
+    #[pallet::getter(fn get_version)]
+    pub(super) type GetVersion<T: Config> = StorageMap<
+        _,
+        Blake2_128Concat,
+        T::ProcessIdentifier,
+        i32,
         ValueQuery
     >;
 
     #[pallet::event]
     #[pallet::generate_deposit(pub(super) fn deposit_event)]
     pub enum Event<T: Config> {
-        // TODO: implement correct events for extrinsics including params
-        ProcessCreated,
-        ProcessDisabled
+        // id, version, restrictions, is_new
+        ProcessCreated(T::ProcessIdentifier, i32, Vec<Restriction>, bool),
+        ProcessDisabled,
     }
 
     #[pallet::error]
     pub enum Error<T> {
         // TODO: implement errors for extrinsics
     }
 
+    // helper functions
+    fn get_version<T: Config>(id: T::ProcessIdentifier) -> i32 {
+        return <GetVersion<T>>::get(&id);
+    }
+
+    fn uopdate_version<T: Config>(id: T::ProcessIdentifier) -> i32 {
+        let version: i32 = get_version::<T>(id.clone());
+        let exists: Result<Process, ()> = <GetProcess<T>>::try_get(id.clone(),version);
+        let new_version: i32 = version + if exists.is_ok() { 0 } else { 1 };
+        <GetVersion<T>>::insert(
```

Comment:
GetVersion::insert feels like conflicting naming here

Reply:
agreed, should rename to just a version and process maybe to keep it simple. Because this allows us to perform a various calls to the storage